src/test/java/com/stripe/model/TopupTest.java

```diff
@@ -0,0 +1,104 @@
+package com.stripe.model;
+
+import com.stripe.BaseStripeTest;
+import com.stripe.exception.StripeException;
+import com.stripe.net.APIResource;
+import com.stripe.net.LiveStripeResponseGetter;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class TopupTest extends BaseStripeTest {
+    @Before
+    public void mockStripeResponseGetter() {
+        APIResource.setStripeResponseGetter(networkMock);
+    }
+
+    @After
+    public void unmockStripeResponseGetter() {
+        /* This needs to be done because tests aren't isolated in Java */
+        APIResource.setStripeResponseGetter(new LiveStripeResponseGetter());
+    }
+
+    @Test
+    public void testDeserialize() throws IOException {
+        String json = resource("topup.json");
+        Topup topup = APIResource.GSON.fromJson(json, Topup.class);
+        assertEquals("tu_HASH", topup.getId());
+        assertEquals("topup", topup.getObject());
+        assertEquals(1000, topup.getAmount().intValue());
+        assertEquals("usd", topup.getCurrency());
+        assertEquals("Testing", topup.getDescription());
+        assertEquals(1518134400, topup.getExpectedAvailabilityDate().intValue());
```

Comment:
Why `intValue` here if `getExpectedAvailabilityDate` is a long?
Additionally, could you do
<pre>
assertEquals(1518134400L, topup.getExpectedAvailabilityDate());
</pre>

Same for `getCreated`?